frontend/controller/app/identity.js

```diff
@@ -311,7 +311,13 @@ export default (sbp('sbp/selectors/register', {
     return sbp('okTurtles.eventQueue/queueEvent', 'APP-LOGIN', async () => {
       console.debug('[gi.app/identity/login] Scheduled call starting', identityContractID, username)
       if (username) {
-        identityContractID = await sbp('namespace/lookup', username)
+        const nsIdentityContractID = await sbp('namespace/lookup', username, { skipCache: process.env.CI || process.env.NODE_ENV !== 'production' })
+        if (!identityContractID) {
+          identityContractID = nsIdentityContractID
+        } else if (nsIdentityContractID !== identityContractID) {
+          console.error(new Error(`Identity contract ID mismatch during login: ${identityContractID} != ${nsIdentityContractID}`))
+          throw new GIErrorUIRuntimeError(L('Identity contract ID mismatch during login'))
+        }
```

Comment:
Why is this new code here? What is it doing? Are you sure it's necessary? If so, the answer to these questions needs to be a comment above line 315

> L('Identity contract ID mismatch during login')

`contract ID` => `contractID`

Reply:
This code is checking that we're logging in to the contract we've just created, which is what was failing before due to a cache issue, but also generally an error from the perspective of the client

> contract ID => contractID

`contract ID` (with a space) is used elsewhere (actually, the exact string). I can change it, but for non-technical users probably contract ID is more meaningful than a made up word like `contractID`

Reply:
> contract ID (with a space) is used elsewhere (actually, the exact string). I can change it, but for non-technical users probably contract ID is more meaningful than a made up word like contractID

Fair enough... although I feel like non-technical users are perfectly capable of picking up such "vocab".

Just a RFC then

Reply:
>  are perfectly capable of picking up such "vocab"

Sure they can, but it's a bit extra cognitive load. Although maybe it's less confusing than thinking about it in terms of a _contract_.